File: leetcode-questions/csharp/src/test/csharp/com/leetcode/questions/projectionareaof3dshapes/ProjectionAreaOf3dShapesTest.cs
using Xunit;
using System.Collections.Generic;
using com.leetcode.questions.leetcode.editor.en.projectionareaof3dshapes;
using com.leetcode.questions.support;

namespace com.leetcode.questions.projectionareaof3dshapes;

public class ProjectionAreaOf3dShapesTest {

    private const string CasesFileName = "projection-area-of-3d-shapes.json";

    public ProjectionAreaOf3dShapesTest() {

    }

    [Fact]
    public void TestCases() {
        GeneratedLeetCodeTestHarness.Run(typeof(Solution), "ProjectionArea", GeneratedLeetCodeTestHarness.ReadCasesFile(CasesFileName));
    }
}

File: leetcode-questions/java/src/test/java/com/leetcode/questions/projectionareaof3dshapes/ProjectionAreaOf3dShapesTest.java
package com.leetcode.questions.projectionareaof3dshapes;

import com.leetcode.questions.leetcode.editor.en.ProjectionAreaOf3dShapes;
import com.leetcode.questions.support.GeneratedLeetCodeTestHarness;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import java.nio.file.Files;
import java.nio.file.Path;

public class ProjectionAreaOf3dShapesTest {

    private static final Path CASES_PATH = Path.of("generated-test-cases", "projection-area-of-3d-shapes.json");

    public ProjectionAreaOf3dShapesTest() {

    }

    @Test
    public void testCases() throws Exception {
        GeneratedLeetCodeTestHarness.run(ProjectionAreaOf3dShapes.Solution.class, "projectionArea", Files.readString(CASES_PATH));
    }
}
